scheduler.py

```diff
@@ -216,7 +220,11 @@ def run_schedule(dryrun=False):
     args = parser.parse_args()
     error_list = Errors()
     
-    run_schedule(args.test)
+    try:
+        run_schedule(args.test)
```

Comment:
It's probably worth moving the `singleton` call from `run_schedule` and in to the main function as you commented.  That way, we'd avoid accidentally picking up a `SystemExit` from somewhere else in `run_schedule`.  That may be unlikely, but generally it's good practice to have as little code running as possible within a try/except block.